tensorflow_addons/metrics/multilabel_metrics.py

```diff
@@ -0,0 +1,235 @@
+import tensorflow as tf
+from tensorflow.keras.metrics import Metric
+
+class MultiLabelMacroRecall(Metric):
+    """Computes the Macro-averaged Recall of the given tensors."""
+
+    def __init__(
+        self,
+        name:str="multi_label_macro_recall",
+        threshold:float=0.5,
+        from_logits:bool=True,
+        activation:str="sigmoid",
+        **kwargs
+    ):
+        """Creates a `MultiLabelMacroRecall` instance.
+
+        Args:
+            name (str): (Optional) string name of the metric instance. Defaults to
+                `"multi_label_macro_recall"`.
+            threshold (float): (Optional) float value to use to binarize labels and
+                predictions. Values greater than `threshold` will set to `1`.
+                Defaults to `0.5`.
+            from_logits (bool): (Optional) boolean value to specifiy whether or not
+                `y_pred` are logits or have not yet been transformed via a final
+                activation function. If `from_logits` is `False`, `activation`
+                will be applied to `y_pred` prior to metric calculation. Defaults
+                to `True`.
+            activation (str): (Optional) string value of the activation function
+                to apply to `y_pred` if `from_logits=False`. Defaults to `"sigmoid"`
+                . Options include `["sigmoid"]`.
+        """
+        super(MultiLabelMacroRecall, self).__init__(name=name, **kwargs)
+        self._threshold = threshold
+        self._from_logits = from_logits
+        self._activation = activation
+        self._recall = self.add_weight(name="mlm_recall", initializer="zeros")
+
+        # NOTE: could be replaced  with tf confusion_matrix utils
+        # whether or not the overhead of that is worth is needs to be tested.
+        self._true_positives  = self.add_weight(name="tp", initializer="zeros")
+        self._false_positives = self.add_weight(name="fp", initializer="zeros")
+
+    def update_state(self, y_true, y_pred, sample_weight=None):
```

Comment:
could you specify the input shape of `y_true` and `y_pred`? I believe it would be helpful for multilabel cases.

Reply:
did so under Recall and as I have seen elsewhere in main repo, Sens/Spec say see Recall for info